shared/ui/combobox/combobox.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { ChangeEvent, FC, MouseEventHandler, useState } from 'react';
+import { clsx } from 'clsx';
+
+import { Combobox as ComboboxUI } from '@headlessui/react';
+import * as ScrollArea from '@radix-ui/react-scroll-area';
+
+import style from './combobox.module.scss';
+
+import { Typography } from '..';
+
+import ArrowDown from '../../../assets/icons/arrow-down.svg?url';
+import Close from '../../../assets/icons/close.svg?url';
+
+import Image from 'next/image';
+
+type Option = {
+  label: string;
+  value: number | string;
+};
+
+export type ComboboxProps = {
+  disabled?: boolean;
+  errorMessage?: string;
+  inputValue: string;
+  label?: string;
+  name?: string;
+  onChange: (value: null | string) => void;
+  onClear?: () => void;
+  onInputChange: (value: string) => void;
+  options: Option[];
+  placeholder?: string;
+  showClearButton?: boolean;
+  value: string;
+};
+
+export const Combobox: FC<ComboboxProps> = props => {
+  const {
+    disabled,
+    errorMessage,
+    inputValue,
+    label,
+    name,
+    onChange,
+    onClear,
+    onInputChange,
+    options,
+    placeholder,
+    showClearButton = true,
+    value,
+  } = props;
+
+  const showError = !!errorMessage && errorMessage.length > 0;
+
+  const inputChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
As mentioned in [Headless UI docs](https://headlessui.com/react/combobox#allowing-empty-values), there is a `nullable` prop that does exactly the same job as your `inputChangeHandler` callback. There is also no need for the state containing the input value to be placed at a higher level. outside the combobox. The state can be placed inside it. Therefore, the `inputValue`, `onInputChange` and `inputChangeHanlder` can be deleted.